File: Original Artifact/src/Task/TaskService.java
package Task;
import java.util.HashMap;

import Task.Task;

public class TaskService {
	
	// Creating an empty HashMap
    private static HashMap<String, Task> task_map = new HashMap<String, Task>();
    
    public static void addTask(String id, String name, String description) {
    	Task task = new Task(id,name,description);
    	if(task_map.containsKey(id)) {
			throw new IllegalArgumentException("Id already exists");
		}
    	task_map.put(id, task);
		return;
    }
    public static void deleteTask(String id) {
    	if(!task_map.containsKey(id)) {
			throw new IllegalArgumentException("Id doesn't exists");
		}
    	task_map.remove(id);
    	return;
    }
    public static void setTaskName(String id, String name) {
    	if(!task_map.containsKey(id)) {
			throw new IllegalArgumentException("Id doesn't exists");
		}
    	task_map.get(id).setName(name);
    	return;
    }
    public static void setTaskDescription(String id, String description) {
    	if(!task_map.containsKey(id)) {
			throw new IllegalArgumentException("Id doesn't exists");
		}
    	task_map.get(id).setName(description);
    	return;
    }
    public static void clearmemory() {
    	task_map.clear();
    	return;
    }
			

}

File: Enhanced Artifact/task_service.py
# -*- coding: utf-8 -*-
"""
File: TaskService.py
Author: Paul Stephan
Version: 1.0
Date: Sep 22, 2024

Description:
This class provides services for managing tasks. It allows adding, updating, and deleting tasks.
"""

from task import Task

class TaskService:
    # Dictionary to store tasks with their IDs as keys
    task_map = {}

    # Method to add a new task to the task map
    @staticmethod
    def add_task(task_id, name, description):
        # Create a new task object
        task = Task(task_id, name, description)

        # Check if the task ID already exists in the map
        if task_id in TaskService.task_map:
            raise ValueError("ID already exists")

        # Add the task to the map
        TaskService.task_map[task_id] = task

    # Method to delete a task from the task map
    @staticmethod
    def delete_task(task_id):
        # Check if the task ID exists in the map
        if task_id not in TaskService.task_map:
            raise ValueError("ID doesn't exist")

        # Remove the task from the map
        del TaskService.task_map[task_id]

    # Method to update the name of a task
    @staticmethod
    def set_task_name(task_id, name):
        # Check if the task ID exists in the map
        if task_id not in TaskService.task_map:
            raise ValueError("ID doesn't exist")

        # Update the name of the task
        TaskService.task_map[task_id].set_name(name)

    # Method to update the description of a task
    @staticmethod
    def set_task_description(task_id, description):
        # Check if the task ID exists in the map
        if task_id not in TaskService.task_map:
            raise ValueError("ID doesn't exist")

        # Update the description of the task
        TaskService.task_map[task_id].set_description(description)

    # Method to clear the task map (empty it)
    @staticmethod
    def clear_memory():
        TaskService.task_map.clear()